testdata/iter8.tpl

```diff
@@ -34,7 +34,29 @@ metadata:
   name: {{ .Name }}-{{ $suffix }}
 stringData:
   experiment: |
-{{ . | toYAML | indent 4 }}
+    # task 1: generate HTTP requests for https://example.com
```

Comment:
Was it the intention to unroll this here. It now differs from the experiment.yaml (cf. run commands 3 and 4 have changed)?

Reply:
Yes. To be super clear....

`iter8 gen go`  does not require any `experiment.yaml`. It is simply a go  template, and simply executes the template with data  suppplied using `--set`. In other words, you can render any string using this command. This is something we plan to use for code engine (for the time being  at least).

`iter8 gen exp` actually renders an `experiment.yaml` file using an `experiment.tpl` file. Only difference between this and`iter8 gen go` is that it will validate that the output generated through this command is a valid experiment spec. Without this validation, this command is equivalent to the above.

`iter8 manifest` has built a built  in template for generating kubernetes manifests. You are creating this!

`iter8 report` produces reports from experiment results.

***

All of the above have some overlapping template and validation functionality (which have been abstracted into reusable pieces).  But they do very different things from end-user perspective (at the minimum, calling for separate documentation).


The idea is `iter8 gen go` will not require an `experiment.yaml` file to be present at all.